File: reconnect_app/lib/features/auth/data/models/login_response.dart
class UserDto {
  final String id;
  final String email;
  final String? username;
  final String role;
  final bool isAnonymous;

  UserDto({
    required this.id,
    required this.email,
    this.username,
    required this.role,
    required this.isAnonymous,
  });

  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] ?? '',
      email: json['email'] ?? '',
      username: json['username'],
      role: json['role'] ?? '',
      isAnonymous: json['isAnonymous'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'username': username,
      'role': role,
      'isAnonymous': isAnonymous,
    };
  }
}

class LoginResponse {
  final UserDto user;
  final String accessToken;
  final String refreshToken;
  final String? accessTokenExpiresAt;
  final String? refreshTokenExpiresAt;
  final int? expiresIn;
  final int? refreshExpiresIn;

  LoginResponse({
    required this.user,
    required this.accessToken,
    required this.refreshToken,
    this.accessTokenExpiresAt,
    this.refreshTokenExpiresAt,
    this.expiresIn,
    this.refreshExpiresIn,
  });

  String get token => accessToken;

  bool get hasRefreshToken => refreshToken.isNotEmpty;

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      user: UserDto.fromJson(json['user']),
      accessToken: json['accessToken']?.toString() ?? json['token']?.toString() ?? '',
      refreshToken: json['refreshToken']?.toString() ?? '',
      accessTokenExpiresAt: json['accessTokenExpiresAt']?.toString(),
      refreshTokenExpiresAt: json['refreshTokenExpiresAt']?.toString(),
      expiresIn: (json['expiresIn'] as num?)?.toInt(),
      refreshExpiresIn: (json['refreshExpiresIn'] as num?)?.toInt(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'user': user.toJson(),
      'accessToken': accessToken,
      'refreshToken': refreshToken,
      'accessTokenExpiresAt': accessTokenExpiresAt,
      'refreshTokenExpiresAt': refreshTokenExpiresAt,
      'expiresIn': expiresIn,
      'refreshExpiresIn': refreshExpiresIn,
      'token': accessToken,
    };
  }
}

File: reconnect_backend/src/main/java/com/reconnect/mindhealth/modules/auth/dto/LoginResponse.java
package com.reconnect.mindhealth.modules.auth.dto;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LoginResponse {
    private UserDto user;
    private String accessToken;
    private String refreshToken;
    private Date accessTokenExpiresAt;
    private Date refreshTokenExpiresAt;
    private Long expiresIn;
    private Long refreshExpiresIn;
    private String token;

    public UserDto getUser() { return user; }
    public void setUser(UserDto user) { this.user = user; }
    public String getAccessToken() { return accessToken; }
    public void setAccessToken(String accessToken) { this.accessToken = accessToken; }
    public String getRefreshToken() { return refreshToken; }
    public void setRefreshToken(String refreshToken) { this.refreshToken = refreshToken; }
    public Date getAccessTokenExpiresAt() { return accessTokenExpiresAt; }
    public void setAccessTokenExpiresAt(Date accessTokenExpiresAt) { this.accessTokenExpiresAt = accessTokenExpiresAt; }
    public Date getRefreshTokenExpiresAt() { return refreshTokenExpiresAt; }
    public void setRefreshTokenExpiresAt(Date refreshTokenExpiresAt) { this.refreshTokenExpiresAt = refreshTokenExpiresAt; }
    public Long getExpiresIn() { return expiresIn; }
    public void setExpiresIn(Long expiresIn) { this.expiresIn = expiresIn; }
    public Long getRefreshExpiresIn() { return refreshExpiresIn; }
    public void setRefreshExpiresIn(Long refreshExpiresIn) { this.refreshExpiresIn = refreshExpiresIn; }
    public String getToken() { return token; }
    public void setToken(String token) { this.token = token; }
}
